cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1066,17 +1067,15 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         }
 
         Future<ReconciliationState> kafkaInitClusterRoleBinding() {
-            ClusterRoleBindingOperator.ClusterRoleBinding desired = kafkaCluster.generateClusterRoleBinding(namespace);
-            Future<Void> fut = clusterRoleBindingOperator.reconcile(
-                    KafkaCluster.initContainerClusterRoleBindingName(namespace, name),
-                    desired);
+            KubernetesClusterRoleBinding desired = kafkaCluster.generateClusterRoleBinding(namespace);
+            Future<ReconcileResult<KubernetesClusterRoleBinding>> fut = clusterRoleBindingOperator.reconcile(
+                    KafkaCluster.initContainerClusterRoleBindingName(namespace, name), desired);
 
             Future replacementFut = Future.future();
 
             fut.setHandler(res -> {
                 if (res.failed()) {
-                    if (desired == null && res.cause().getMessage().contains("403: Forbidden")) {
-                        log.debug("Ignoring forbidden access to ClusterRoleBindings which seems not needed while Kafka rack awareness is disabled.");
```

Comment:
Should we still keep this message? It might get us some insights into what is going on.

Reply:
Issue is we got this message from WorkaroundOperator which operated with http requests. With Kubernetes object the reason of exception may differ.

Reply:
You're right that the client will throw an exception, but what happens in the 403 case? Are we just doing to propagate the exception? Shouldn't we catch it and log like we are doing here?

Reply:
I am not sure what do you mean. If the exception has some message it is logged 
```
if (res.cause() != null) {
       debugMsg += " " + res.cause().getMessage();
}
```

Reply:
Can't we detect RBAC issues form the exception? Because the code above is IMHO wrong ... without the 403 you do not really know whether the rack awareness is disabled. The original idea was if I did not had RBAC in the first place, I couldn't have enabled it in the first place - without the RBAC check this does not fully apply.